app/Http/Controllers/SessionController.php

```diff
@@ -167,7 +165,7 @@ public function login(LoginRequest $request): void
 	 */
 	public function logout(): void
```

Comment:
Logout on the SessionController is called from `api/Session::logout` and takes care of the return codes etc.
Logout from `Authorization` is also called from tests and Livewire.

Reply:
Sorry, my fault. The original question was rather dump. I did not realize that I was looking at a HTTP controller.

But there is more to it. As discussed in https://github.com/LycheeOrg/Lychee/pull/1403#discussion_r922688180 the controller method should actually look like

```php
public function logout(Request $request): void {
    Auth::logout();
     $request->session()->invalidate();
     $request->session()->regenerateToken();
}
```

It somehow makes sense to invalidate the HTTP session in a HTTP controller. However, this code has been moved to the specialized `Auth::logout` method which is not only called by this method, but also in tests etc. I wonder if it correct to also fiddle with the session in the tests.

I wonder what would happen (especially what would break), if we only had this code here and only called the umodified `Auth::login` and `Auth::logout` (i.e. of the parent class) everywhere else. In this case we would not need to overwrite `Auth::logout` in our own custom `Auth` class.

Reply:
@ildyria Do you have an idea, i.e. answer to these questions?

> I wonder if it correct to also fiddle with the session in the tests.

> I wonder what would happen (especially what would break), if we only had this code here and only called the umodified Auth::login and Auth::logout (i.e. of the parent class) everywhere else.

Reply:
Didn't take much time to think about that one as I was heavily refactoring the other part.

Reply:
@nagmat84 can this be marked as resolved as I nuked the Authorization::logout ? :D